### DISCUSSION TITLE
Set Fixed Tiles for Generator to Work Around

### DISCUSSION THREAD

**Discussion by Supiyo** [AlexeyBond/godot-constraint-solving#20] [Q&A] [answered]
Hello, thank you so much for making this plugin! It is easy to understand how to use it, but I have one question regarding what the title says. I have a single tile that I want to be set every time shown in the picture. In the how to, it says, _Create a TileMap or GridMap generated map will be written to. The new map should use the same tile set/mesh library as one created on step 2. You may place some tiles on that map (either manually or procedurally), generator will take them into account and fill other cells accordingly. But try to not create an unsolvable puzzle when doing so._ However, instead of working around that tile in the target tilemap, the generator finishes very quickly and nothing more other than that tile is outputted.

This is my first time asking a question on a forum like platform for coding and this question might have not made sense, so if any clarification is needed, please do not hesitate to ask.

![image](https://github.com/AlexeyBond/godot-constraint-solving/assets/109705481/2d6181f9-5aa8-454f-9350-8017132f47d7)

**Comment by AlexeyBond** [comment, marked as answer]
Hi. Does it work without the fixed tile? Can you show the generator node settings and logs from "output" tab after running the generator?

**Reply by Supiyo** [reply]
Hi, thank you so much for your quick reply. Upon looking at the output tab, it said failed backtracking and that made me realize that I forgot to put some much needed layered tiles around that tile shown in the picture in the positive sample. After adding those tiles in, it does correctly take the tile in account and generate around it. Bit of a silly mistake on my part but thank you for helping me nonetheless. I might have a question later though about caller the .start() method is the assert(_runner = null) in the generator_2d,gd script keeps failing, but I haven't looked at it too extensively yet. 

Again, Thank you!

![image](https://github.com/AlexeyBond/godot-constraint-solving/assets/109705481/7e52e9ee-24c2-4d45-b9ae-56f23f91f280)
runner is no longer null after the first call of start()

**Reply by AlexeyBond** [reply]
As mentioned in the comment on the screenshot, the `start` method must not be called twice (and should not be called at all if autostart is enabled). If you really want to start generator again - call `reset` before doing so. If you want to run multiple generations concurrently - use multiple generator nodes.

**Reply by Supiyo** [reply]
Ah, I saw that it couldn't be called twice, but just didn't know what to do. Reset works perfectly.

Thank you so much for all your help and making this plugin!
